# Charles claims that he can distinguish between beer and ale 75 percent of the time. 

# Ruth bets that he cannot and, in fact, just guesses. To settle this, a bet is made: Charles is to be given ten small glasses, each having been filled with beer or ale, chosen by tossing a fair coin. 

# He wins the bet if he gets seven or more correct. 

### a) Find the probability that Charles wins if he has the ability that he claims. 

### b) Find the probability that Ruth wins if Charles is guessing

____
# a)

## We can use the BinomialProbabilities function with n = 10, p = 0.7, kmin = 7 and kmax = 10

In [1]:
from math import factorial

def choose(n, r):
    numerator = float(factorial(n))
    denominator = factorial(r)*factorial(n-r)
    
    return numerator/denominator

def BinomialProbabilities(n, p, kmin, kmax):
    list_values = []
    for k in range(kmin, kmax+1):
        val = choose(n, k)*((p)**k)*((1-p)**(n-k))
        list_values.append(val)
    return(sum(list_values))

In [2]:
BinomialProbabilities(10, 0.7, 7, 10)

0.6496107183999998

## So if he actually is as good as he says he is, he has a 65% chance of winning the bet

___
# b)

## We can use the BinomialProbabilities function with n = 10, p = 0.5, kmin = 7 and kmax = 10, then take 1- the result

In [3]:
1-BinomialProbabilities(10, 0.5, 7, 10)

0.828125

## So if he's just guessing, she has a 83% chance of winning the bet